homeassistant/components/lg_thinq/climate.py

```diff
@@ -0,0 +1,344 @@
+    step: float | None = None
+
+
+DEVIE_TYPE_CLIMATE_MAP: dict[DeviceType, tuple[ThinQClimateEntityDescription, ...]] = {
+    DeviceType.AIR_CONDITIONER: (
+        ThinQClimateEntityDescription(
+            key=ExtendedProperty.CLIMATE_AIR_CONDITIONER,
+            name=None,
+            translation_key=ExtendedProperty.CLIMATE_AIR_CONDITIONER,
+        ),
+    ),
+    DeviceType.SYSTEM_BOILER: (
+        ThinQClimateEntityDescription(
+            key=ExtendedProperty.CLIMATE_SYSTEM_BOILER,
+            name=None,
+            translation_key=ExtendedProperty.CLIMATE_SYSTEM_BOILER,
+            min_temp=16,
+            max_temp=30,
+            step=1,
+        ),
+    ),
+}
+
+STR_TO_HVAC: dict[str, HVACMode] = {
+    "air_dry": HVACMode.DRY,
+    "auto": HVACMode.AUTO,
+    "cool": HVACMode.COOL,
+    "fan": HVACMode.FAN_ONLY,
+    "heat": HVACMode.HEAT,
+}
+
+HVAC_TO_STR: dict[HVACMode, str] = {
+    HVACMode.AUTO: "auto",
+    HVACMode.COOL: "cool",
+    HVACMode.DRY: "air_dry",
+    HVACMode.FAN_ONLY: "fan",
+    HVACMode.HEAT: "heat",
+}
+
+THINQ_PRESET_MODE: list[str] = ["air_clean", "aroma", "energy_saving"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ThinqConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up an entry for climate platform."""
+    entities: list[ThinQClimateEntity] = []
+    for coordinator in entry.runtime_data.values():
+        if (
+            descriptions := DEVIE_TYPE_CLIMATE_MAP.get(
+                coordinator.api.device.device_type
+            )
+        ) is not None:
+            for description in descriptions:
+                entities.extend(
+                    ThinQClimateEntity(coordinator, description, property_id)
+                    for property_id in coordinator.api.get_active_idx(description.key)
+                )
+
+    if entities:
+        async_add_entities(entities)
+
+
+class ThinQClimateEntity(ThinQEntity, ClimateEntity):
+    """Represent a thinq climate platform."""
+
+    entity_description: ThinQClimateEntityDescription
+
+    def __init__(
+        self,
+        coordinator: DeviceDataUpdateCoordinator,
+        entity_description: ThinQClimateEntityDescription,
+        property_id: str,
+    ) -> None:
+        """Initialize a climate entity."""
+        super().__init__(coordinator, entity_description, property_id)
+
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.TURN_ON
+            | ClimateEntityFeature.TURN_OFF
+        )
+        self._attr_hvac_modes = [HVACMode.OFF]
+        self._attr_hvac_mode = HVACMode.OFF
+        self._attr_fan_modes = []
+        self._attr_fan_mode = None
+        self._attr_preset_modes = []
+        self._attr_preset_mode = None
+        self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        self._requested_hvac_mode: str | None = None
```

Comment:
_attr_temperature_unit has to be initialized in constructor() because runtime error can be caused in super().__init__().

Reply:
`super.__init__()` doesn't contain any references to this right?

Reply:
Since we haven't defined for temperature_unit's property , we need this statement for initialization.
Otherwise, the following error occurs.
2024-09-19 10:17:21.937 ERROR (MainThread) [homeassistant.components.climate] Error adding entity None for domain climate with platform lg_thinq
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 598, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 816, in _async_add_entity
    capabilities=entity.capability_attributes,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 323, in __getattribute__
    return super().__getattribute__(__name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 489, in capability_attributes
    temperature_unit = self.temperature_unit
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 323, in __getattribute__
    return super().__getattribute__(__name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/functools.py", line 995, in __get__
    val = self.func(instance)
          ^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 579, in temperature_unit
    return self._attr_temperature_unit
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 323, in __getattribute__
    return super().__getattribute__(__name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 323, in __getattribute__
    return super().__getattribute__(__name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ThinQClimateEntity' object has no attribute '__attr_temperature_unit'. Did you mean: '_attr_temperature_unit'?